### DISCUSSION TITLE
Allow playback, timeline, etc whilst waveform renders in background

### DISCUSSION THREAD

**Discussion by WilliamVenner** [katspaugh/wavesurfer.js#3871] [Q&A] [answered]
Hi all,

Is it possible to render the waveform "in the background" and still allow the user to see a timeline, seek, interact with the audio etc.?

I have a very large audio file and it can take some time to render the waveform (in the event that the server has not yet pre-generated the peaks and it must be done locally); but I would like the user to still be able to play the audio and seek it.

**Comment by katspaugh** [comment, marked as answer]
You can render your audio track in a simple audio element and connect that to wavesurfer. See an example here: https://wavesurfer.xyz/examples/?video.js
